### PR TITLE
Add typescript types

### LINKED ISSUE #8: TypeScript support
It would be nice if this project had TypeScript types

### PR DESCRIPTION
**Pull request by lenovouser**
Closes #8

### CODE REVIEW

**Comment by lenovouser** [comment]
Ping @huan @simonepri

**Comment by simonepri** [comment]
@lenovouser sorry this went off my radar thanks for the ping.

**Comment by simonepri** [comment]
@lenovouser
Can you fix the typing to match [these guidelines](https://github.com/sindresorhus/typescript-definition-style-guide)? (Ignore the one about using tabs and use 2 space indentation instead)
You can find a [good example here](https://github.com/sindresorhus/execa/blob/master/index.d.ts)

Thanks!

**Comment by lenovouser** [comment]
@simonepri no problem. Should be done!

**Comment by lenovouser** [comment]
@simonepri ping 😃

**Comment by lenovouser** [comment]
@simonepri I think this is ready to merge 🚀

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ node_js:
   - '8'
   - '6'
   - '4'
+matrix:
+  allow_failures:
+  - node_js: '4'
```
Comment (soyuka) [inline .travis.yml]: Does it fail on node 4? Not a fan of allowing failures for it, or we need to remove it from the supported version (major version bump).

Comment (lenovouser) [reply .travis.yml]: Yes it does, but unrelated to this PR (master is already failing).

I didn't want to remove it in this PR to keep it focused on the TS changes.

Comment (soyuka) [reply .travis.yml]: I see, only the linter fails, please remove this from this PR and we're going to patch this in another one thanks!

**Comment by soyuka** [comment]
tyvm @lenovouser !

**Pull request merged**